service/matching/task_reader.go

```diff
@@ -235,10 +235,16 @@ func (tr *taskReader) getTaskBatchWithRange(
 	return response.Tasks, err
 }
 
+type tasksBatch struct {
+	Tasks           []*persistencespb.AllocatedTaskInfo
+	ReadLevel       int64
+	IsReadBatchDone bool
+}
```

Comment:
nit: getTasksBatchResponse maybe? read level and the done flag look like metadata to the returned batch of tasks.

do those fields have to be exported?

Reply:
Renamed and made them unexported